src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSchemaJsonWriter.cs

```diff
@@ -0,0 +1,1292 @@
+
+            // whose value is an object.
+            this.jsonWriter.WriteStartObject();
+
+            // The function import object MUST contain the member $Kind with a string value of FunctionImport
+            this.jsonWriter.WriteRequiredProperty("$Kind", CsdlConstants.Element_FunctionImport);
+
+            // The function import object MUST contain the member $Function.
+            this.jsonWriter.WriteRequiredProperty("$Function", functionImport.Operation.FullName());
+
+            // The function import object MAY contain the member $EntitySet.
+            this.WriteOperationImportAttributes(functionImport, CsdlConstants.Attribute_Function);
+
+            // The value of $IncludeInServiceDocument is one of the Boolean literals true or false. Absence of the member means false.
+            if (functionImport.IncludeInServiceDocument)
+            {
+                this.jsonWriter.WriteRequiredProperty("$IncludeInServiceDocument", functionImport.IncludeInServiceDocument);
+            }
+        }
+
+        internal override void WriteOperationParameterElementHeader(IEdmOperationParameter parameter, bool inlineType)
+        {
+            this.jsonWriter.WriteStartObject();
+
+            // A parameter object MUST contain the member $Name
+            this.jsonWriter.WriteRequiredProperty("$Name", parameter.Name);
+
+            if (inlineType)
+            {
+                WriteTypeReference(parameter.Type);
+            }
+        }
+
+        private void WriteTypeReference(IEdmTypeReference type, string defaultTypeName = "Edm.String")
+        {
+            IEdmTypeReference elementType = type;
+            if (type.IsCollection())
+            {
+                this.jsonWriter.WriteRequiredProperty("$Collection", true);
+
+                IEdmCollectionTypeReference collectionReference = type.AsCollection();
+                elementType = collectionReference.ElementType();
+            }
+
+            // Absence of the $Type member means the type is Edm.String.
+            // Does it mean to omit the type for Collection(Edm.String)? No, $Collection is used to identify whether it's collection of not.
+            if (elementType.FullName() != defaultTypeName)
+            {
+                string typeName = this.SerializationName((IEdmSchemaElement)elementType.Definition);
+                this.jsonWriter.WriteRequiredProperty("$Type", typeName);
+            }
+        }
+
+        internal override void WriteOperationParameterEndElement(IEdmOperationParameter parameter)
+        {
+            IEdmOptionalParameter optionalParameter = parameter as IEdmOptionalParameter;
+            if (optionalParameter != null && !(optionalParameter.VocabularyAnnotations(this.Model).Any(a => a.Term == CoreVocabularyModel.OptionalParameterTerm)))
+            {
+                string defaultValue = optionalParameter.DefaultValueString;
+                EdmRecordExpression optionalValue = new EdmRecordExpression();

```

Comment:
maybe move these into default value conditional statement in 730?

Reply:
Emm, It's used at Line 727. So, keep it as there.

---
In reply to: [443756247](https://github.com/OData/odata.net/pull/1775#discussion_r443756247) [](ancestors = 443756247)

Reply:
right. missed that. thanks.

---
In reply to: [444390242](https://github.com/OData/odata.net/pull/1775#discussion_r444390242) [](ancestors = 444390242,443756247)